# Read in packages
```{r}
require(ggplot2)
require(plyr)
require(lubridate)
require(dplyr)
require(sp)
require(maptools)
require(ggmap)
require(ggplot2)
require(reshape2)
require(rgdal)
require(raster)
require(rgeos)
require(PBSmapping)
require(gdistance)
require(grid)
require(gridExtra)
```

### plotting Norway data
```{r}
adult <- read.table("./results_scen1wild1/adult.txt", header=TRUE)
herit.FW <- read.table("./results_scen1wild1/heritabilityFreshwater.txt", header=TRUE)
herit.S <- read.table("./results_scen1wild1/heritabilitySea.txt", header=TRUE)
smolts <- read.table("./results_scen1wild1/smolts.txt", header=TRUE)
embryo <- read.table("./results_scen1wild1/embryo.txt", header=TRUE)
parr0 <- read.table("./results_scen1wild1/parr0.txt", header=TRUE)
parr1 <- read.table("./results_scen1wild1/parr1.txt", header=TRUE)
parr2 <- read.table("./results_scen1wild1/parr2.txt", header=TRUE)
spawners <- read.table("./results_scen1wild1/spawners.txt", fill=TRUE, colClasses = "character")
colnames(spawners) <- spawners[1,]
spawners <- spawners[2:399,]

##fry <- read.table("./results_scen1wild1/fry.txt", header=TRUE)

juv <- rbind(parr0, parr1, parr2)
juv$YC <- c(0,0,1,1,2,2)

# Can I replicate Figure 2 in the IBSEM paper?

plot1 <- plot(1, embryo$density)
plot2 <- plot(juv$YC, juv$density)


head(adult)

p0 <- ggplot() + geom_line(data=adult[1:250,], aes(month, sw0), colour="red") +
  geom_point(data=adult[1:250,], aes(month, sw0), colour="red") +
  theme_bw()
p1 <- ggplot() + geom_line(data=adult[1:250,], aes(month, sw1), colour="orange") +
  geom_point(data=adult[1:250,], aes(month, sw1), colour="orange") +
  theme_bw()
p2 <- ggplot() + geom_line(data=adult[1:250,], aes(month, sw2), colour="green") +
  geom_point(data=adult[1:250,], aes(month, sw2), colour="green") +
  theme_bw()
p3 <- ggplot() + geom_line(data=adult[1:250,], aes(month, sw3), colour="blue") +
  geom_point(data=adult[1:250,], aes(month, sw3), colour="blue") +  
  theme_bw()

grid.arrange(p0,p1,p2,p3, nrow=4)

ggplot() + geom_line(data=adult[1:250,], aes(month, total)) +
  geom_point(data=adult[1:250,], aes(month, total)) +
  theme_bw()

adult$month.adj <- adult$month%%12

ggplot() + 
  geom_boxplot(data=adult, aes(month.adj, popSize, group=month.adj)) +
  theme_bw()

sept <- subset(adult, month.adj == 9)
mean(sept$sw0)
mean(sept$sw1)
mean(sept$sw2)
mean(sept$sw3)
sd(sept$sw0)
sd(sept$sw1)
sd(sept$sw2)
sd(sept$sw3)


spawners$month.adj <- as.numeric(spawners$month)%%12
octob <- subset(spawners, month.adj==9)
colnames(octob)[9:12] <- c("sw0", "sw1", "sw2", "sw3")
mean(as.numeric(octob$sw3))

variables <- data.frame(vars = c(colnames(adult), colnames(smolts), colnames(parr0), colnames(parr1), colnames(parr2), colnames(spawners), colnames(embryo), colnames(herit.FW), colnames(herit.S)))


pAdult <- read.table("C:/Users/keyserf/Desktop/salmonIBM/Install/src/runFiles/parameters/adultParameters.txt")
```


### read in environmental data
```{r}
require(raster)
require(ncdf4)

temps <- nc_open("C:/Users/keyserf/Downloads/sst.mnmean.v3.nc")
temps <- raster(temps)
lon <- ncvar_get(temps, "lon")
nlon <- dim(lon)
lat <- ncvar_get(temps, "lat")
nlat <- dim(lat)
time <- ncvar_get(temps, "time")
ntime <- dim(time)
meas <- ncvar_get(temps, "sst")

m <- 1
tmp.slice <- meas[, , m]

plot(raster(tmp.slice))

print(temps)

df <- expand.grid(lon, lat, time)
meas.vec <- as.vector(meas)
df <- data.frame(cbind(df, meas.vec))
names(df) <- c("lon", "lat", "dayssince", "temp")

test <- ddply(.data=df, .(lon, lat),
              summarize,
              mnmean=mean(temp, na.rm=TRUE))

ggplot() + geom_point(data=test, aes(lon, lat, colour=mnmean)) +
  geom_rect(data=test, aes(xmin=290, xmax=310, ymin=45, ymax=60), fill=NA, colour="white")

df_atl <- subset(df, lon >290 & lon <310 & lat>45 & lat < 55 & is.na(temp)==FALSE)

head(df_atl)
df_atl$date <- ymd("1800-01-01") + days(df_atl$dayssince)
df_atl$month <- month(df_atl$date)
df_atl$year <- year(df_atl$date)
df_atl <- subset(df_atl, year > 1985)

mnmean <- ddply(.data=df_atl, .(month),
                summarize,
                mntemp = mean(temp))

ggplot() + geom_point(data=mnmean, aes(month, mntemp)) + theme_bw()

### mnmean contains monthly 30-year averaged sea temperatures.

### next steps: need river temperature data
```



### analyse 50-year salmon data to get parameters
### read in table
```{r}
## only some of the data
ian <- read.table("C:/Users/keyserf/Documents/Data/All Biochar NET and RR Oct 2013.txt", header=TRUE)
names(ian)

ian[,1:20] <- apply(ian[,1:20], 2, function(x) as.numeric(as.character(x)))


## sites
sites <- read.csv("C:/Users/keyserf/Documents/Data/sites_lat_long_brendan.csv")

## all the data
salmon <- read.csv("C:/Users/keyserf/Documents/Data/All Biochar Oct 2013.csv", header=TRUE)
names(salmon)

str(salmon)
unique(salmon$SEX)

## correct the year column
salmon$year.adj <- as.numeric(ifelse(salmon$YEAR < 10, paste0(200, salmon$YEAR), ifelse(salmon$YEAR %in% c(10,11,12), paste0(20, salmon$YEAR), salmon$YEAR)))

salmon[,5:28] <- apply(salmon[,5:28], 2, function(x) as.numeric(as.character(x)))

## look at sampling timing
ggplot() + geom_histogram(data=salmon, aes(MO), colour="black", fill="white") + scale_x_continuous(breaks=seq(0, 13,1)) + theme_bw() 

## format dates
salmon$DATE <- ymd(paste0(salmon$year.adj,"-", salmon$MO,"-", salmon$DY))
salmon$dayssince <- salmon$DATE - ymd("1954-01-01")
salmon$doy <- salmon$DATE - ymd(paste0(salmon$year.adj, "-", 1, "-", 1))

ggplot() + geom_histogram(data=salmon[salmon$RA>1,], aes(MO, fill=as.factor(RA)), binwidth=1, colour="black", position=position_dodge()) + scale_x_continuous(breaks=seq(0, 12,1)) + theme_bw() 

### spring parr = 1
### spring smolts = 2
### 1 summer smolts = 3

# RA = river age?
# SA = sea age?


### need to separate by season due to summer growth
salmon$season <- ifelse(salmon$MO %in% c(4,5,6), "spring", ifelse(salmon$MO %in% c(7,8), "summer", ifelse(salmon$MO %in% c(9,10,11), "fall", NA)))

## unique(salmon$AREA) = 48
## unique(salmon$year.adj) = 59
## unique(salmon$MO) = 10 incl. NA
## unique(salmon$DY) = 32 incl. NA
## unique(salmon$GEAR) = 21 incl. NA
## unique(salmon$LS) = 12 incl. NA
## unique(salmon$FL) = 58 incl. NA
## unique(salmon$WW) = >3300 incl. NA
## unique(salmon$SEX) = 7 incl. NA
## unique(salmon$MAT) = 5 incl. NA
## unique(salmon$RA) = 13 incl. NA
## unique(salmon$VSA) = 5 incl. NA
## unique(salmon$TSM) = 8 incl. NA
## unique(salmon$SSA1) = 4 incl. NA
## unique(salmon$SSA2) = 3 incl. NA
## unique(salmon$SSA3) = 3 incl. NA
## unique(salmon$SSA4) = 2 incl. NA
## unique(salmon$SA) = 8 incl. NA and 0? landlocked or?
## unique(salmon$TA) = 15 incl. NA. 25 (1), 11 (5), 14 (1) too...
## unique(salmon$EDGE) = 8 incl. NA
## unique(salmon$REL) = 8 incl. NA
## unique(salmon$TEGGS) = >1870 incl. NA


##clean up by decade
salmon$decade <- substr(salmon$year.adj, 3, 3)
salmon$decade <- as.factor(salmon$decade)
salmon$decade <- factor(salmon$decade, levels=c("5", "6", "7", "8", "9", "0", "1"))

# fifties <- subset(salmon, decade==5)
# sixties <- subset(salmon, decade==6)
# seventies <- subset(salmon, decade==7)
# eighties <- subset(salmon, decade==8)
# nineties <- subset(salmon, decade==9)
# millen <- subset(salmon, decade==0)
# teens <- subset(salmon, decade==1)
# 
# boxplot(fifties$FL)
# boxplot(sixties$FL)
# boxplot(seventies$FL)
# boxplot(eighties$FL)
# boxplot(nineties$FL)
# boxplot(millen$FL)
# boxplot(teens$FL)
# 
# ggplot() + geom_point(data=fifties, aes(FL, WW, colour=season), alpha=0.5) + facet_wrap(~AREA) + theme_bw() + theme(panel.grid=element_blank())
# ggplot() + geom_point(data=sixties, aes(FL, WW, colour=season), alpha=0.5) + facet_wrap(~AREA) + theme_bw() + theme(panel.grid=element_blank())
# ggplot() + geom_point(data=seventies, aes(FL, WW, colour=season), alpha=0.5) + facet_wrap(~AREA) + theme_bw() + theme(panel.grid=element_blank())
# ggplot() + geom_point(data=eighties, aes(FL, WW, colour=season), alpha=0.5) + facet_wrap(~AREA) + theme_bw() + theme(panel.grid=element_blank())
# ggplot() + geom_point(data=nineties, aes(FL, WW, colour=season), alpha=0.5) + facet_wrap(~AREA) + theme_bw() + theme(panel.grid=element_blank())
# ggplot() + geom_point(data=millen, aes(FL, WW, colour=season), alpha=0.5) + facet_wrap(~AREA) + theme_bw() + theme(panel.grid=element_blank())
# ggplot() + geom_point(data=teens, aes(FL, WW, colour=season), alpha=0.5) + facet_wrap(~AREA) + theme_bw() + theme(panel.grid=element_blank())

salmon$WWcorr <- salmon$WW/10
#unique(salmon_clean$AREA)
salmon_clean <- subset(salmon, is.na(MO)==FALSE & is.na(FL)==FALSE & TA < 12 & AREA %in% c(sites$X.1, 341079))
salmon_clean <- subset(salmon_clean, REL>2 | is.na(REL)==TRUE)
#salmon_NAs <- subset(salmon_clean, is.na(REL)==TRUE)
#salmon_best <- subset(salmon_clean, REL>3)

#ggplot() + geom_point(data=salmon_clean, aes(log(FL), log(WWcorr), colour=test)) + geom_smooth(data=salmon_clean, aes(log(FL), log(WWcorr)), method="lm") + facet_wrap(~TA)

salmon_clean$test <- ifelse(salmon_clean$TA==4 & log(salmon_clean$FL) >6 & log(salmon_clean$WWcorr) <5, "bad",
                            ifelse(salmon_clean$TA==5 & log(salmon_clean$FL) >6 & log(salmon_clean$WWcorr) <6.25, "bad",
                                   ifelse(salmon_clean$TA==6 & log(salmon_clean$FL) >6 & log(salmon_clean$WWcorr) <6.5, "bad", "ok")))
#table(salmon_clean$test)
salmon_clean <- subset(salmon_clean, !test=="bad")

#ggplot() + geom_point(data=salmon_clean, aes(TA, FL, colour=test)) + geom_smooth(data=salmon_clean, aes(TA, FL), method="lm") + facet_wrap(~LS)

salmon_clean$season <- factor(salmon_clean$season, levels=c("spring", "summer", "fall"))

# ggplot() + geom_histogram(data=salmon_clean[salmon_clean$FL < 500,], aes(FL, fill=as.factor(AREA)), binwidth=20) + facet_grid(TA~season)
# ggplot() + geom_histogram(data=salmon_clean[salmon_clean$FL > 500,], aes(FL, fill=as.factor(AREA)), binwidth=20) + facet_grid(TA~season)
# 
# ggplot() + geom_boxplot(data=salmon_clean, aes(decade, FL))
# ggplot() + geom_boxplot(data=salmon_clean, aes(decade, WWcorr))
# 
# ggplot() + geom_point(data=salmon_NAs, aes(log(FL), log(WWcorr), colour=season)) + facet_grid(~decade) +
#   geom_smooth(data=salmon_NAs, aes(log(FL), log(WWcorr), colour=season), method="lm")
```


### only look at fish caught as smolts
```{r}
smolts <- subset(salmon_clean, LS ==2)

### get rid of outliers
# ggplot() + geom_point(data=smolts, aes(FL, WW))
# ggplot() + geom_boxplot(data=smolts, aes(group=as.factor(TA), TA, FL))

smolts$season <- factor(smolts$season, levels=c("spring", "summer", "fall"))

smolts$scaledFL <- base::scale(smolts$FL)
```

### plot smolt lengths against year and by nyears in river 
```{r}
plot1 <- ggplot() + 
  geom_point(data=smolts[smolts$season=="spring" & is.na(smolts$season)==FALSE,], aes(year.adj, scaledFL), size=0.5) + 
  facet_grid(RA~.) +
  theme_bw() +
  geom_smooth(data=smolts[smolts$season=="spring" & is.na(smolts$season)==FALSE,], aes(year.adj, scaledFL), method="lm") + 
  ylab("Scaled fork length") +
  xlab("Year")

model1 <- lm(data=smolts[which(smolts$RA==2 & smolts$season=="spring"),], scaledFL~year.adj,na.action = na.exclude)
summary(model1)

model2 <- lm(data=smolts[which(smolts$RA==3 & smolts$season=="spring"),], scaledFL~year.adj)
summary(model2)

model3 <- lm(data=smolts[which(smolts$RA==4 & smolts$season=="spring"),], scaledFL~year.adj)
summary(model3)

model4 <- lm(data=smolts[which(smolts$RA==5 & smolts$season=="spring"),], scaledFL~year.adj)
summary(model4)

vals <- data.frame(year.adj=rep(1985,4), scaledFL=rep(7, 4), RA=c(2,3,4,5),
                   slope = c(coefficients(model1)[2],coefficients(model2)[2],coefficients(model3)[2],coefficients(model4)[2]),
                   rsq = c(summary(model1)$adj.r.squared, summary(model2)$adj.r.squared, summary(model3)$adj.r.squared, summary(model4)$adj.r.squared))

plot1 + geom_text(data=vals, aes(year.adj, scaledFL, label=paste0("slope = ", round(slope,4), "   adj. Rsq. = ", round(rsq,4))), size=3, hjust=0) + ylim(-3, 8)

##############################
```


## Calculating natural mortality
```{r}
salmon_clean$sex.adj <- ifelse(salmon_clean$SEX %in% c(1, 3, 4), "M", ifelse(salmon_clean$SEX %in% c(2,5), "F", NA))

agestr <- ddply(.data=salmon_clean[!is.na(salmon_clean$SEX)==TRUE,], .(TA, sex.adj),
                summarize,
                total = length(FL))

ggplot() + geom_point(data=agestr, aes(TA, log(total))) + facet_wrap(~sex.adj) +
  geom_smooth(data=agestr[agestr$TA>3,], aes(TA, log(total)), method="lm") +
  theme_classic() + theme(panel.border=element_rect(colour="black", fill=NA))

model1 <- lm(data=agestr[agestr$TA >4 & agestr$sex.adj=="M",], log(total)~TA) 
summary(model1)
model2 <- lm(data=agestr[agestr$TA >4 & agestr$sex.adj=="F",], log(total)~TA) 
summary(model2)

# total mortality for males = -2.4299 | for females = -2.22645
# error for males = 0.1343 | for females = 0.06711

# exp(-2.4299) = 0.08804564 survival per year males = 8.8% survival?
# exp(-2.22645) = 0.1079108 survival per year females =10.7% survival?

mortF4 <- log(3583) - log(687)
mortF5 <- log(687) - log(66)
mortF6 <- log(66) - log(8)
mortM4 <- log(1216) - log(258)
mortM5 <- log(258) - log(18)
mortM6 <- log(18) - log(2)

rates <- data.frame(rate=rbind(mortF4, mortF5, mortF6, mortM4, mortM5, mortM6))
rates <- data.frame(cbind(round(rates,3), sex=c(rep("F",3), rep("M", 3)), age=c(4,5,6,4,5,6)))

rates$surv <- round(exp(-rates$rate),3)
rates
## survival per year for each age/sex combo
# rate sex age  surv
# 1.652   F   4 0.192
# 2.343   F   5 0.096
# 2.110   F   6 0.121
# 1.550   M   4 0.212
# 2.197   M   5 0.111
# 2.197   M   6 0.111


### ok but this is not age-specific mortality. Papers recommend using dry weight to predict age-specific mortality, so need to calculate dry weight from length or from wet weight. For NL parr: log10(dryweight) = -2.43 + 2.76*log10(Length) OR log10(dryweight) = -2.43 + 2.76*(1.76 + log10(wetweight))/2.78
parr <- subset(salmon_clean, LS==1)
parr$DW1 <- exp(-2.43 + 2.76*log10(parr$FL/10))

ageweight <- ddply(.data=parr, .(RA),
                   summarize,
                   meanwt = mean(DW1))

# McGurk 1986 weight-mortality relationship: Mw = 0.00526Wexp(-0.25)
ageweight$Mw <- 0.00526*ageweight$meanwt^-0.25


#smolting probability = LS 3 / LS 2
966/5129 
# = 0.1883

#maturation probability = number of mature/number available

max(table(salmon$AREA))
site <- table(salmon$AREA)

site <- melt(site)
site <- subset(site, value> 10000)

unique(site$Var1)
```


### try Piou and Prevost IBASAM package
```{r}
## see github page: https://github.com/Ibasam/IBASAM

require(Ibasam)
install.packages("Ibasam")
update.packages(checkBuilt=TRUE, ask=FALSE)

library(devtools)
install("C:/Users/keyserf/Documents/Data/Ibasam/Compiled/Ibasam_2.1.0.zip")

```


### FOR PARR PRESENTATION
```{r}
## example plots
## Figure 9 - a,b,c from paper (scenario 3: Low, medium, high)
adult <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/adult.txt", header=TRUE)
heritabilityFreshwater <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/heritabilityFreshwater.txt", header=TRUE)
heritabilitySea <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/heritabilitySea.txt", header=TRUE)
smolts <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/smolts.txt",header=TRUE)
embryo <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/embryo.txt", fill=TRUE, colClasses = "character")
parr0 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/parr0.txt", header=TRUE)
parr1 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/parr1.txt", header=TRUE)
parr2 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/parr2.txt", header=TRUE)
spawners <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/spawners.txt", fill=TRUE, colClasses = "character")

colnames(spawners) <- spawners[1,]
spawners <- spawners[2:399,]
colnames(embryo) <- embryo[1,]
embryo <- embryo[2:399,]
embryo[,c(1,30:31)] <- apply(embryo[,c(1,30:31)], 2, function(x) as.numeric(x))
embryo <- subset(embryo, is.na(genotypicValueAverage)=="FALSE")
adult <- subset(adult, is.na(genotypicValueAverage)=="FALSE")
smolts <- subset(smolts, is.na(genotypicValueAverage)=="FALSE")

labels <- data.frame(month=c(5000, 5000, 5000), genotypicValueAverage=c(0.93, 0.89, 0.8), text=c("Adult", "Smolt", "Embryo"))

## genetic effects (low introgression, scenario 3)
plot9a1 <- ggplot() + 
  geom_ribbon(data=embryo, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25) +
  geom_ribbon(data=adult, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25) +
  geom_ribbon(data=smolts, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25)+
  geom_point(data=adult, aes(month, genotypicValueAverage), shape=1) +
  geom_point(data=embryo, aes(month, genotypicValueAverage), shape=4) + 
  geom_point(data=smolts, aes(month, genotypicValueAverage), shape=2) +
  geom_text(data=labels, aes(month, genotypicValueAverage, label=text)) +
  xlim(600,5500) +
  ylim(0.7, 1)+
  xlab("Month") +
  ylab("Average genotypic value") +
  theme_classic() +
  theme(panel.background=element_rect(colour="black"))

colnames(spawners)[9:12] <- c("SW0", "SW1", "SW2", "SW3")

spawners[,c(1:51)] <- apply(spawners[,c(1:51)], 2, function(x) as.numeric(x))


### adult spawner population size (low introgression, scenario 3)
spawn_popsize <- spawners[,c(1,9:12)]
spawn_popsize <- melt(data=spawn_popsize, id.vars=c("month"))

plot9a2 <- ggplot() + 
  geom_point(data=spawn_popsize[!spawn_popsize$variable =="SW0",], aes(month, value))+
  facet_wrap(~variable, nrow=3) +
  xlab("Month") +
  ylab("Population size") +
  theme_classic() +
  theme(panel.background=element_rect(colour="black"))
                  
grid.arrange(plot9a1, plot9a2, ncol=2, top="Low introgression, standard parameters")


### plot 7a1-3
str(embryo)
embryo[,9:29] <- apply(embryo[,9:29], 2, function(x) as.numeric(x))
embryoplot <- dplyr::select(embryo, month, 9:29)
embryoplot <- melt(embryoplot, id.vars="month")
embryoplot <- subset(embryoplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
embryoplot$variable <- as.factor(as.character(embryoplot$variable))

levels(embryoplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=embryoplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")
  
embryoplot$timebin[embryoplot$month <1200] <- "50-100"
embryoplot$timebin[embryoplot$month >1199 & embryoplot$month <2400] <- "100-200"
embryoplot$timebin[embryoplot$month >2399 & embryoplot$month <3600] <- "200-300"
embryoplot$timebin[embryoplot$month >3599 & embryoplot$month <4800] <- "300-400"
embryoplot$timebin[embryoplot$month >4799 & embryoplot$month <6000] <- "400-500"

embryoplot$timebin <- factor(embryoplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

embryomean <- ddply(.data=embryoplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7emb <- ggplot() + geom_boxplot(data=embryoplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=embryomean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") +
  ggtitle("Embryos")

# smolt
### plot 7a1-3
str(smolts)
smolts[,25:45] <- apply(smolts[,25:45], 2, function(x) as.numeric(x))
smoltsplot <- dplyr::select(smolts, month, 25:45)
smoltsplot <- melt(smoltsplot, id.vars="month")
smoltsplot <- subset(smoltsplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
smoltsplot$variable <- as.factor(as.character(smoltsplot$variable))

levels(smoltsplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=smoltsplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")
  
smoltsplot$timebin[smoltsplot$month <1200] <- "50-100"
smoltsplot$timebin[smoltsplot$month >1199 & smoltsplot$month <2400] <- "100-200"
smoltsplot$timebin[smoltsplot$month >2399 & smoltsplot$month <3600] <- "200-300"
smoltsplot$timebin[smoltsplot$month >3599 & smoltsplot$month <4800] <- "300-400"
smoltsplot$timebin[smoltsplot$month >4799 & smoltsplot$month <6000] <- "400-500"

smoltsplot$timebin <- factor(smoltsplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

smoltsmean <- ddply(.data=smoltsplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7smolts <- ggplot() + geom_boxplot(data=smoltsplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=smoltsmean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") +
  ggtitle("Smolts")

# adult
### plot 7a1-3
str(adult)
adult[,30:50] <- apply(adult[,30:50], 2, function(x) as.numeric(x))
adultplot <- dplyr::select(adult, month, 30:50)
adultplot <- melt(adultplot, id.vars="month")
adultplot <- subset(adultplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
adultplot$variable <- as.factor(as.character(adultplot$variable))

levels(adultplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=adultplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")

adultplot$timebin[adultplot$month <1200] <- "50-100"
adultplot$timebin[adultplot$month >1199 & adultplot$month <2400] <- "100-200"
adultplot$timebin[adultplot$month >2399 & adultplot$month <3600] <- "200-300"
adultplot$timebin[adultplot$month >3599 & adultplot$month <4800] <- "300-400"
adultplot$timebin[adultplot$month >4799 & adultplot$month <6000] <- "400-500"

adultplot$timebin <- factor(adultplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

adultmean <- ddply(.data=adultplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7adults <- ggplot() + geom_boxplot(data=adultplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=adultmean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") + 
  ggtitle("Adults")

grid.arrange(plot7emb,plot7smolts, plot7adults, nrow=3, top="Low introgression, standard parameters")

### plot 9B medium introgression

adult <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/adult.txt", header=TRUE)
heritabilityFreshwater <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/heritabilityFreshwater.txt", header=TRUE)
heritabilitySea <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/heritabilitySea.txt", header=TRUE)
smolts <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/smolts.txt",header=TRUE)
embryo <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/embryo.txt", fill=TRUE, colClasses = "character")
parr0 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/parr0.txt", header=TRUE)
parr1 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/parr1.txt", header=TRUE)
parr2 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/parr2.txt", header=TRUE)
spawners <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/spawners.txt", fill=TRUE, colClasses = "character")

colnames(spawners) <- spawners[1,]
spawners <- spawners[2:399,]
colnames(embryo) <- embryo[1,]
embryo <- embryo[2:399,]
embryo[,c(1,30:31)] <- apply(embryo[,c(1,30:31)], 2, function(x) as.numeric(x))
embryo <- subset(embryo, is.na(genotypicValueAverage)=="FALSE")
adult <- subset(adult, is.na(genotypicValueAverage)=="FALSE")
smolts <- subset(smolts, is.na(genotypicValueAverage)=="FALSE")

labels <- data.frame(month=c(5000, 5000, 5000), genotypicValueAverage=c(0.925, 0.825, 0.73), text=c("Adult", "Smolt", "Embryo"))

plot9b1 <- ggplot() + 
  geom_ribbon(data=embryo, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25) +
  geom_ribbon(data=adult, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25) +
  geom_ribbon(data=smolts, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25)+
  geom_point(data=adult, aes(month, genotypicValueAverage), shape=1) +
  geom_point(data=embryo, aes(month, genotypicValueAverage), shape=4) + 
  geom_point(data=smolts, aes(month, genotypicValueAverage), shape=2) +
  geom_text(data=labels, aes(month, genotypicValueAverage, label=text)) +
  xlim(600,5500) +
  ylim(0.6, 1)+
  xlab("Month") +
  ylab("Average genotypic value") +
  theme_classic() +
  theme(panel.background=element_rect(colour="black"))

colnames(spawners)[9:12] <- c("SW0", "SW1", "SW2", "SW3")

spawners[,c(1:51)] <- apply(spawners[,c(1:51)], 2, function(x) as.numeric(x))


### adult spawner population size (med introgression, scenario 3)
spawn_popsize <- spawners[,c(1,9:12)]
spawn_popsize <- melt(data=spawn_popsize, id.vars=c("month"))

plot9b2 <- ggplot() + 
  geom_point(data=spawn_popsize[!spawn_popsize$variable =="SW0",], aes(month, value))+
  facet_wrap(~variable, nrow=3) +
  xlab("Month") +
  ylab("Population size") +
  theme_classic() +
  theme(panel.background=element_rect(colour="black"))
                  
grid.arrange(plot9b1, plot9b2, ncol=2, top="Medium introgression, standard parameters")



### plot 7a1-3
str(embryo)
embryo[,9:29] <- apply(embryo[,9:29], 2, function(x) as.numeric(x))
embryoplot <- dplyr::select(embryo, month, 9:29)
embryoplot <- melt(embryoplot, id.vars="month")
embryoplot <- subset(embryoplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
embryoplot$variable <- as.factor(as.character(embryoplot$variable))

levels(embryoplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=embryoplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")
  
embryoplot$timebin[embryoplot$month <1200] <- "50-100"
embryoplot$timebin[embryoplot$month >1199 & embryoplot$month <2400] <- "100-200"
embryoplot$timebin[embryoplot$month >2399 & embryoplot$month <3600] <- "200-300"
embryoplot$timebin[embryoplot$month >3599 & embryoplot$month <4800] <- "300-400"
embryoplot$timebin[embryoplot$month >4799 & embryoplot$month <6000] <- "400-500"

embryoplot$timebin <- factor(embryoplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

embryomean <- ddply(.data=embryoplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7emb <- ggplot() + geom_boxplot(data=embryoplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=embryomean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") +
  ggtitle("Embryos")

# smolt
### plot 7a1-3
str(smolts)
smolts[,25:45] <- apply(smolts[,25:45], 2, function(x) as.numeric(x))
smoltsplot <- dplyr::select(smolts, month, 25:45)
smoltsplot <- melt(smoltsplot, id.vars="month")
smoltsplot <- subset(smoltsplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
smoltsplot$variable <- as.factor(as.character(smoltsplot$variable))

levels(smoltsplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=smoltsplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")
  
smoltsplot$timebin[smoltsplot$month <1200] <- "50-100"
smoltsplot$timebin[smoltsplot$month >1199 & smoltsplot$month <2400] <- "100-200"
smoltsplot$timebin[smoltsplot$month >2399 & smoltsplot$month <3600] <- "200-300"
smoltsplot$timebin[smoltsplot$month >3599 & smoltsplot$month <4800] <- "300-400"
smoltsplot$timebin[smoltsplot$month >4799 & smoltsplot$month <6000] <- "400-500"

smoltsplot$timebin <- factor(smoltsplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

smoltsmean <- ddply(.data=smoltsplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7smolts <- ggplot() + geom_boxplot(data=smoltsplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=smoltsmean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") +
  ggtitle("Smolts")

# adult
### plot 7a1-3
str(adult)
adult[,30:50] <- apply(adult[,30:50], 2, function(x) as.numeric(x))
adultplot <- dplyr::select(adult, month, 30:50)
adultplot <- melt(adultplot, id.vars="month")
adultplot <- subset(adultplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
adultplot$variable <- as.factor(as.character(adultplot$variable))

levels(adultplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=adultplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")

adultplot$timebin[adultplot$month <1200] <- "50-100"
adultplot$timebin[adultplot$month >1199 & adultplot$month <2400] <- "100-200"
adultplot$timebin[adultplot$month >2399 & adultplot$month <3600] <- "200-300"
adultplot$timebin[adultplot$month >3599 & adultplot$month <4800] <- "300-400"
adultplot$timebin[adultplot$month >4799 & adultplot$month <6000] <- "400-500"

adultplot$timebin <- factor(adultplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

adultmean <- ddply(.data=adultplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7adults <- ggplot() + geom_boxplot(data=adultplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=adultmean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") + 
  ggtitle("Adults")

grid.arrange(plot7emb,plot7smolts, plot7adults, nrow=3, top="Medium introgression, standard parameters")




### plot 9C high introgression

adult <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/adult.txt", header=TRUE)
heritabilityFreshwater <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/heritabilityFreshwater.txt", header=TRUE)
heritabilitySea <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/heritabilitySea.txt", header=TRUE)
smolts <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/smolts.txt",header=TRUE)
embryo <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/embryo.txt", fill=TRUE, colClasses = "character")
parr0 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/parr0.txt", header=TRUE)
parr1 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/parr1.txt", header=TRUE)
parr2 <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/parr2.txt", header=TRUE)
spawners <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/spawners.txt", fill=TRUE, colClasses = "character")

colnames(spawners) <- spawners[1,]
spawners <- spawners[2:399,]
colnames(embryo) <- embryo[1,]
embryo <- embryo[2:399,]
embryo[,c(1,30:31)] <- apply(embryo[,c(1,30:31)], 2, function(x) as.numeric(x))
embryo <- subset(embryo, is.na(genotypicValueAverage)=="FALSE")
adult <- subset(adult, is.na(genotypicValueAverage)=="FALSE")
smolts <- subset(smolts, is.na(genotypicValueAverage)=="FALSE")

labels <- data.frame(month=c(200, 200, 200), genotypicValueAverage=c(0.925, 0.85, 0.77), text=c("Adult", "Smolt", "Embryo"))

plot9c1 <- ggplot() + 
  geom_ribbon(data=embryo, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25) +
  geom_ribbon(data=adult, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25) +
  geom_ribbon(data=smolts, aes(month, ymax=genotypicValueAverage+(genotypicValueStDev/sqrt(21)), ymin=genotypicValueAverage-(genotypicValueStDev/sqrt(21))), fill="black", alpha=0.25)+
  geom_point(data=adult, aes(month, genotypicValueAverage), shape=1) +
  geom_point(data=embryo, aes(month, genotypicValueAverage), shape=4) + 
  geom_point(data=smolts, aes(month, genotypicValueAverage), shape=2) +
  geom_text(data=labels, aes(month, genotypicValueAverage, label=text)) +
  xlim(0,5500) +
  ylim(0, 1)+
  xlab("Month") +
  ylab("Average genotypic value") +
  theme_classic() +
  theme(panel.background=element_rect(colour="black"))

colnames(spawners)[9:12] <- c("SW0", "SW1", "SW2", "SW3")

spawners[,c(1:51)] <- apply(spawners[,c(1:51)], 2, function(x) as.numeric(x))


### adult spawner population size (med introgression, scenario 3)
spawn_popsize <- spawners[,c(1,9:12)]
spawn_popsize <- melt(data=spawn_popsize, id.vars=c("month"))

plot9c2 <- ggplot() + 
  geom_point(data=spawn_popsize[!spawn_popsize$variable =="SW0",], aes(month, value))+
  facet_wrap(~variable, nrow=3) +
  xlab("Month") +
  ylab("Population size") +
  theme_classic() +
  theme(panel.background=element_rect(colour="black"))
                  
grid.arrange(plot9c1, plot9c2, ncol=2, top="High introgression, standard parameters")



### plot 7a1-3
str(embryo)
embryo[,9:29] <- apply(embryo[,9:29], 2, function(x) as.numeric(x))
embryoplot <- dplyr::select(embryo, month, 9:29)
embryoplot <- melt(embryoplot, id.vars="month")
embryoplot <- subset(embryoplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
embryoplot$variable <- as.factor(as.character(embryoplot$variable))

levels(embryoplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=embryoplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")
  
embryoplot$timebin[embryoplot$month <1200] <- "50-100"
embryoplot$timebin[embryoplot$month >1199 & embryoplot$month <2400] <- "100-200"
embryoplot$timebin[embryoplot$month >2399 & embryoplot$month <3600] <- "200-300"
embryoplot$timebin[embryoplot$month >3599 & embryoplot$month <4800] <- "300-400"
embryoplot$timebin[embryoplot$month >4799 & embryoplot$month <6000] <- "400-500"

embryoplot$timebin <- factor(embryoplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

embryomean <- ddply(.data=embryoplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7emb <- ggplot() + geom_boxplot(data=embryoplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=embryomean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") +
  ggtitle("Embryos")

# smolt
### plot 7a1-3
str(smolts)
smolts[,25:45] <- apply(smolts[,25:45], 2, function(x) as.numeric(x))
smoltsplot <- dplyr::select(smolts, month, 25:45)
smoltsplot <- melt(smoltsplot, id.vars="month")
smoltsplot <- subset(smoltsplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
smoltsplot$variable <- as.factor(as.character(smoltsplot$variable))

levels(smoltsplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=smoltsplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")
  
smoltsplot$timebin[smoltsplot$month <1200] <- "50-100"
smoltsplot$timebin[smoltsplot$month >1199 & smoltsplot$month <2400] <- "100-200"
smoltsplot$timebin[smoltsplot$month >2399 & smoltsplot$month <3600] <- "200-300"
smoltsplot$timebin[smoltsplot$month >3599 & smoltsplot$month <4800] <- "300-400"
smoltsplot$timebin[smoltsplot$month >4799 & smoltsplot$month <6000] <- "400-500"

smoltsplot$timebin <- factor(smoltsplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

smoltsmean <- ddply(.data=smoltsplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7smolts <- ggplot() + geom_boxplot(data=smoltsplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=smoltsmean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") +
  ggtitle("Smolts")

# adult
### plot 7a1-3
str(adult)
adult[,30:50] <- apply(adult[,30:50], 2, function(x) as.numeric(x))
adultplot <- dplyr::select(adult, month, 30:50)
adultplot <- melt(adultplot, id.vars="month")
adultplot <- subset(adultplot, variable %in% c("allele_0_0_Frequency",
                                                 "allele_0_4_Frequency",
                                                 "allele_0_9_Frequency",
                                                 "allele_0_14_Frequency",
                                                 "allele_0_19_Frequency",
                                                 "allele_0_20_Frequency"))
adultplot$variable <- as.factor(as.character(adultplot$variable))

levels(adultplot$variable) <- c("1", "5", "10", "15", "20", "21") 

ggplot() + geom_line(data=adultplot, aes(month, value, colour=variable))+
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Month") +
  ylab("Allele frequency") +
  scale_colour_discrete(name="Gene")

adultplot$timebin[adultplot$month <1200] <- "50-100"
adultplot$timebin[adultplot$month >1199 & adultplot$month <2400] <- "100-200"
adultplot$timebin[adultplot$month >2399 & adultplot$month <3600] <- "200-300"
adultplot$timebin[adultplot$month >3599 & adultplot$month <4800] <- "300-400"
adultplot$timebin[adultplot$month >4799 & adultplot$month <6000] <- "400-500"

adultplot$timebin <- factor(adultplot$timebin, levels=c("50-100", 
                                                          "100-200",
                                                          "200-300",
                                                          "300-400",
                                                          "400-500"))

adultmean <- ddply(.data=adultplot, .(timebin, variable),
                    summarize,
                    value = mean(value))

plot7adults <- ggplot() + geom_boxplot(data=adultplot, aes(timebin, value, fill=variable), position=position_dodge(width=0.25)) +
  geom_line(data=adultmean, aes(timebin, value, group=variable, colour=variable), linetype="dashed")+
  #facet_wrap(~variable) +
  theme_classic() +
  theme(panel.background=element_rect(colour="black")) +
  xlab("Year span") +
  ylab("Allele frequency") +
  scale_fill_discrete(name="Gene") +
  scale_colour_discrete(name="Gene") + 
  ggtitle("Adults")

grid.arrange(plot7emb,plot7smolts, plot7adults, nrow=3, top="High introgression, standard parameters")

```

### scenario 3 adult pop summaries (low, med, high)
```{r}
adultlo <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 25 - same strayers 5%/wild - 1/runFiles/results/adult.txt", header=TRUE)
adultme <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 100 - same strayers 5%/wild - 1/runFiles/results/adult.txt", header=TRUE)
adulthi <- read.table("C:/Users/keyserf/Documents/Data/IBSEM data/temp/scenario 3/farm 250 - same strayers 5%/wild - 1/runFiles/results/adult.txt", header=TRUE)

head(adultlo)
adultlo <- adultlo[,1:29]
adultme <- adultme[,1:29]
adulthi <- adulthi[,1:29]

str(adultlo)

adultlo$timebin[adultlo$month <1200] <- "50-100"
adultlo$timebin[adultlo$month >1199 & adultlo$month <2400] <- "100-200"
adultlo$timebin[adultlo$month >2399 & adultlo$month <3600] <- "200-300"
adultlo$timebin[adultlo$month >3599 & adultlo$month <4800] <- "300-400"
adultlo$timebin[adultlo$month >4799 & adultlo$month <6000] <- "400-500"

adultme$timebin[adultme$month <1200] <- "50-100"
adultme$timebin[adultme$month >1199 & adultme$month <2400] <- "100-200"
adultme$timebin[adultme$month >2399 & adultme$month <3600] <- "200-300"
adultme$timebin[adultme$month >3599 & adultme$month <4800] <- "300-400"
adultme$timebin[adultme$month >4799 & adultme$month <6000] <- "400-500"

adulthi$timebin[adulthi$month <1200] <- "50-100"
adulthi$timebin[adulthi$month >1199 & adulthi$month <2400] <- "100-200"
adulthi$timebin[adulthi$month >2399 & adulthi$month <3600] <- "200-300"
adulthi$timebin[adulthi$month >3599 & adulthi$month <4800] <- "300-400"
adulthi$timebin[adulthi$month >4799 & adulthi$month <6000] <- "400-500"

adultlosumm <- ddply(.data=adultlo, .(timebin),
                     summarize,
                     avgpopsize = mean(popSize),
                     avgmeanage = mean(averageAge),
                     avgdensity = mean(density),
                     avgL = mean(averageLength),
                     avgW = mean(averageWeight))

adultlosumm <- cbind(adultlosumm, intro=rep("lo",5))

adultmesumm <- ddply(.data=adultme, .(timebin),
                     summarize,
                     avgpopsize = mean(popSize),
                     avgmeanage = mean(averageAge),
                     avgdensity = mean(density),
                     avgL = mean(averageLength),
                     avgW = mean(averageWeight))

adultmesumm <- cbind(adultmesumm, intro=rep("me",5))

adulthisumm <- ddply(.data=adulthi, .(timebin),
                     summarize,
                     avgpopsize = mean(popSize),
                     avgmeanage = mean(averageAge),
                     avgdensity = mean(density),
                     avgL = mean(averageLength),
                     avgW = mean(averageWeight))

adulthisumm <- cbind(adulthisumm, intro=rep("hi",5))

adultsumm <- rbind(adultlosumm, adultmesumm, adulthisumm)

adultlo <- cbind(adultlo, intro=rep("lo", 995))
adultme <- cbind(adultme, intro=rep("me", 995))
adulthi <- cbind(adulthi, intro=rep("hi", 995))

adultraw <- rbind(adultlo, adultme, adulthi)

ggplot() + geom_smooth(data=adultraw, aes(month, averageLength, colour=intro)) 
  #geom_point(data=adultraw, aes(month, averageLength, colour=intro))

ggplot() + geom_smooth(data=adultraw, aes(month, averageWeight, colour=intro)) 
  #geom_point(data=adultraw, aes(month, averageWeight, colour=intro))

ggplot() + geom_smooth(data=adultraw, aes(month, popSize, colour=intro))
  #geom_point(data=adultraw, aes(month, popSize, colour=intro))

ggplot() + geom_smooth(data=adultraw, aes(month, density, colour=intro))

ggplot() + geom_smooth(data=adultraw, aes(month, averageAge, colour=intro))
